Cron/LogCleaner.php

```diff
@@ -0,0 +1,116 @@
+ * It is available through the world-wide-web at this URL:
+ * https://tldrlegal.com/license/mit-license
+ * If you are unable to obtain it through the world-wide-web, please send an email
+ * to support@buckaroo.nl so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future. If you wish to customize this module for your
+ * needs please contact support@buckaroo.nl for more information.
+ *
+ * @copyright Copyright (c) Buckaroo B.V.
+ * @license   https://tldrlegal.com/license/mit-license
+ */
+namespace Buckaroo\Magento2\Cron;
+
+use Magento\Framework\EntityManager\MetadataPool;
+use Buckaroo\Magento2\Model\ConfigProvider\Account;
+use Magento\Framework\App\ResourceConnection;
+use Buckaroo\Magento2\Logging\Log;
+use Buckaroo\Magento2\Model\Config\Source\LogHandler;
+use Magento\Framework\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem\Driver\File;
+use Magento\Framework\Filesystem\Io\File as IoFile;
+
+class LogCleaner
+{
+    private $resource;
+    private $accountConfig;
+    private $resourceConnection;
+    private $logging;
+    private $directoryList;
+    private $driverFile;
+    private $ioFile;
+
+    public function __construct(
+        \Buckaroo\Magento2\Model\ResourceModel\Log $resource,
+        Account $accountConfig,
+        ResourceConnection $resourceConnection,
+        Log $logging,
+        DirectoryList $directoryList,
+        File $driverFile,
+        IoFile $ioFile
+    ) {
+        $this->resource = $resource;
+        $this->accountConfig       = $accountConfig;
+        $this->resourceConnection  = $resourceConnection;
+        $this->logging             = $logging;
+        $this->directoryList       = $directoryList;
+        $this->driverFile          = $driverFile;
+        $this->ioFile              = $ioFile;
+    }
+    
+    public function execute()
+    {
+        $retentionPeriod = (int) $this->accountConfig->getLogRetention();
+        $logHandlerType = (int) $this->accountConfig->getLogHandler();
+
+        if ($retentionPeriod) {
+            if ($logHandlerType == LogHandler::TYPE_DB) {
+                $this->proceedDb($retentionPeriod);
+            }
+            if ($logHandlerType == LogHandler::TYPE_FILES) {
+                $this->proceedFiles($retentionPeriod);
+            }
+        }
+        return $this;
+    }
+
+    private function proceedDb(int $retentionPeriod)
+    {
+        $this->logging->addDebug(__METHOD__ . '|1|' . var_export($retentionPeriod, true));
+
+        $this->resourceConnection->getConnection()->delete(
+            $this->resource->getMainTable(),
+            ['time <= date_sub(now(),interval ' . $retentionPeriod . ' second)']
+        );
+    }
+
+    private function proceedFiles(int $retentionPeriod)
+    {
+        $this->logging->addDebug(__METHOD__ . '|1|' . var_export($retentionPeriod, true));
```

Comment:
To be removed, no added value

Reply:
I think when you will start to work with support tasks and experience hundreds of installations and different behavior at them then you will understand that logging helps a lot to answer when/what/how was executed and to answer/solve customer requests, but no problem - it will be removed.